lib/Runtime/Library/JavascriptError.cpp

```diff
@@ -694,7 +694,7 @@ namespace Js
         return false;
     }
 
-    bool JavascriptError::ThrowCantDelete(PropertyOperationFlags flags, ScriptContext* scriptContext, PCWSTR varName)
+    bool JavascriptError::ThrowCantDeleteIfStrictModeOrNonconfigurable(PropertyOperationFlags flags, ScriptContext* scriptContext, PCWSTR varName)
```

Comment:
>ThrowCantDeleteIfStrictModeOrNonconfigurable [](start = 26, length = 44)

Ugly name, but similar to "ThrowCantDeleteIfStrictMode".

Reply:
`ThrowCantDeleteIfRequiredForNonConfigurableProperty`? Current name sounds like it would throw if the property was nonconfigurable (regardless of the operation flags), which isn't true (and in fact this method has no way to even know if the property was configurable, which must be checked by the caller).

---
In reply to: [173516752](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173516752) [](ancestors = 173516752)

Reply:
Also, on the topic of this method, I don't like how it forces the caller to do a possibly-expensive conversion to string before figuring out whether we need it. I'd rather take a Var and do `GetPropertyDisplayNameForError(varName, scriptContext)->GetString()` only once we determine that we actually need to throw. Throwing is super expensive anyway, so any additional conversion cost there doesn't particularly matter. Anyhow, this wasn't introduced by your change, so feel free to leave as-is.

---
In reply to: [173520340](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173520340) [](ancestors = 173520340,173516752)

Reply:
1. name: it throws either in strict mode or if non-configurable. Because this information can only be extracted from the flags, I think the name is more clear now (it doesn't give the impression that it guarantees to throw anymore).
2. string vs var: the callers seem to differ in how they get hold of the string and I don't have a good grasp of details to assess whether var would make perf better for them.

---
In reply to: [173521221](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173521221) [](ancestors = 173521221,173520340,173516752)

Reply:
Would be fair to have an overload of this method which takes a `Var` for `varName` and then use `GetPropertyDisplayNameForError` if we'll throw an error. Otherwise don't do the conversion. Could save some time in that case.